checker/src/abstract_value.rs

```diff
@@ -332,6 +339,7 @@ pub trait AbstractValueTrait: Sized {
     fn equals(&self, other: Self) -> Self;
     fn greater_or_equal(&self, other: Self) -> Self;
     fn greater_than(&self, other: Self) -> Self;
+    fn has_tag(&self, tag: &Tag) -> Option<Rc<AbstractValue>>;
```

Comment:
I find myself wondering why the return result is optional. It seems that a lifted Boolean can always be returned.